fixed wrong body been used when simple traffic body doesnt exist

Please-Pick-a-Name opened the pull request:
before this, when the simple vehicle option is turned on, and if there is an entry in simplified_vehicles for the vehicle, it will attempt to replace the body slot of said vehicle to the simple traffic body
regardless if the part actually exist:
- instead of the original body, vehicles end up with the default body, if the part doesn't exist or not valid
added bonus:
since the recent beamng update, the naming convention for the simple traffic parts have changed, that means:
- all vehicles, include the one with the simple traffic variant, will have their body changed to default body

i want to fix it so it check if the part exists and the slot matches, and i want to update the list in simplified_vehicles

so far i have done the existence check, but idk how to test it, someone plz tell me beammp deving workflow

Please-Pick-a-Name committed:
make sure the simple traffic part exist

Please-Pick-a-Name committed:
Update MPVehicleGE.lua

sync check to both veh edit and spawn
now check for part slot type (not slots2 compatible)

Please-Pick-a-Name committed:
updated simplified_vehicles list

my feeling towards their new naming convention is mixed

Please-Pick-a-Name commented:
it should now check both if the part exist, and if the part's slot type matches/contains where it is expected to fit in
also updated the simplified_vehicles list
didnt test, dunno how to test

Please-Pick-a-Name commented:
fixed title cause i cannot english

Please-Pick-a-Name commented:
ik how to test my pr now...
yes it errored out, fixing rn

Please-Pick-a-Name committed:
Update MPVehicleGE.lua

- actually requires the jbeam/io
- use getAvailableSlotMap instead of getPart from jbeamio (former check for slots2, and the latter just dont handle stuff well)
- check if the entry exist first
- tested and works

Please-Pick-a-Name commented:
actually figured out beammp contribute workflow, this "works_on_my_machine.txt"
thanks o1ler

Starystars67 reviewed (dismissed):
Testing this I had no errors or issues.
Thank you

Please-Pick-a-Name commented:
dead pr kewk

20dka commented on lua/ge/extensions/MPVehicleGE.lua:
```diff
@@ -895,8 +905,16 @@ local function applyVehEdit(serverID, data)
 
-	if settings.getValue("simplifyRemoteVehicles") and simplified_vehicles[vehicleName] then
-		vehicleConfig.parts[vehicleName..'_body'] = simplified_vehicles[vehicleName]
+	if settings.getValue("simplifyRemoteVehicles") then
```
Could you extract the contents of the if check to a separate function to avoid duplicated code?

Please-Pick-a-Name replied:
sure, its now in vehicleGetSimplified instead

20dka commented on lua/ge/extensions/MPVehicleGE.lua:
```diff
@@ -845,8 +847,16 @@ local function applyVehSpawn(event)
+		local expectedPartID=simplified_vehicles[vehicleName]
+		if expectedPartID then
+			local ioCtx={preloadedDirs={string.format("/vehicles/%s/",vehicleName),"/vehicles/common/"}} -- Fake io context for jbeamIO
+			local slotMap=jbeamIO.getAvailableSlotMap(ioCtx) -- slots2 compatible, maybe theres better way then require jbeamio and let it load the files again...
```
```suggestion
			local slotMap = jbeamIO.getAvailableSlotMap(ioCtx) -- slots2 compatible, maybe theres better way then require jbeamio and let it load the files again...
```
Please leave whitespace around assignment statements. Do this everywhere in your code

Please-Pick-a-Name replied:
done

Please-Pick-a-Name committed:
Improved implementation

added getVehicleSimplified() for checking and getting vehicle's simplified body slot/part name
and spaces ;-;
